doc/example/model_selection.ipynb

```diff
@@ -0,0 +1,828 @@
+    "| M1_2     | example_modelSelection.yaml | 0.2      | estimate | 0        |\n",
+    "| M1_3     | example_modelSelection.yaml | estimate | 0.1      | 0        |\n",
+    "| M1_4     | example_modelSelection.yaml | 0.2      | estimate | estimate |\n",
+    "| M1_5     | example_modelSelection.yaml | estimate | 0.1      | estimate |\n",
+    "| M1_6     | example_modelSelection.yaml | estimate | estimate | 0        |\n",
+    "| M1_7     | example_modelSelection.yaml | estimate | estimate | estimate |"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Forward Selection, Multiple Searches\n",
+    "Here, we show a typical workflow for model selection. First, a [PEtab Select](https://github.com/petab-dev/petab_select) problem is created, which is used to initialize a pyPESTO model selection problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pypesto.select\n",
+    "import petab_select\n",
+    "from petab_select import Criterion, Method, ESTIMATE\n",
+    "petab_select_problem = petab_select.Problem.from_yaml('model_selection/petab_select_problem.yaml')\n",
+    "\n",
+    "\n",
+    "# Helpers for plotting\n",
+    "def get_labels(models):\n",
+    "    labels = {\n",
+    "        model.get_hash(): str(model.model_subspace_id)\n",
+    "        for model in models\n",
+    "    }\n",
+    "    return labels\n",
+    "\n",
+    "\n",
+    "def get_digraph_labels(models, criterion):\n",
+    "    zero = min(model.get_criterion(criterion) for model in models)\n",
+    "    labels = {\n",
+    "        model.get_hash(): f'{model.model_subspace_id}\\n{model.get_criterion(criterion) - zero:.2f}'\n",
+    "        for model in models\n",
+    "    }\n",
+    "    return labels"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "import logging\n",
+    "import pypesto.logging\n",
+    "pypesto.logging.log(level=logging.INFO, name='pypesto.select', console=True)\n",
+    "import petab\n",
+    "\n",
+    "pypesto_select_problem_1 = pypesto.select.Problem(petab_select_problem=petab_select_problem)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Models can be selected with a model selection algorithm (here: [forward](https://en.wikipedia.org/wiki/Stepwise_regression)) and a comparison criterion (here: [AIC](https://en.wikipedia.org/wiki/Akaike_information_criterion)). The forward method involves iterations where each iteration tests all models with one additional estimated parameter.\n",
+    "\n",
+    "As no initial model is specified here, a virtual initial model with no estimated parameters is automatically used to find the \"smallest\" (in terms of number of estimated parameters) models. In this example, this is the model `M1_0`, which has no estimated parameters."
```

Comment:
So each call to `select()` ups the number of parameters by 1? Is there a function that iterates over all model parameter numbers (i.e. until convergence) automatically?

Reply:
Now documented `select_to_completion()` more clearly in the notebook, after the 4th code cell.